stdlib/LinearAlgebra/test/uniformscaling.jl

```diff
@@ -82,24 +139,48 @@ end
 let
     λ = complex(randn(),randn())
     J = UniformScaling(λ)
-    @testset "transpose, conj, inv" begin
+    @testset "transpose, conj, inv, pinv, cond" begin
         @test ndims(J) == 2
         @test transpose(J) == J
         @test J * [1 0; 0 1] == conj(*(adjoint(J), [1 0; 0 1])) # ctranpose (and A(c)_mul_B)
         @test I + I === UniformScaling(2) # +
         @test inv(I) == I
         @test inv(J) == UniformScaling(inv(λ))
+        @test pinv(J) == UniformScaling(inv(λ))
+        @test @inferred(pinv(0.0I)) == 0.0I
+        @test @inferred(pinv(0I)) == 0.0I
+        @test @inferred(pinv(false*I)) == 0.0I
+        @test @inferred(pinv(0im*I)) == 0im*I
         @test cond(I) == 1
         @test cond(J) == (λ ≠ zero(λ) ? one(real(λ)) : oftype(real(λ), Inf))
     end
 
+    @testset "real, imag, reim" begin
+        @test real(J) == UniformScaling(real(λ))
+        @test imag(J) == UniformScaling(imag(λ))
+        @test reim(J) == (UniformScaling(real(λ)), UniformScaling(imag(λ)))
+    end
+
     @testset "copyto!" begin
         A = Matrix{Int}(undef, (3,3))
         @test copyto!(A, I) == one(A)
         B = Matrix{ComplexF64}(undef, (1,2))
         @test copyto!(B, J) == [λ zero(λ)]
     end
 
+    @testset "binary ops with vectors" begin
+        v = complex.(randn(3), randn(3))
+        # As shown in #20423@GitHub, vector acts like x1 matrix when participating in linear algebra
+        @test all(v  * J .== v  * λ)
+        @test all(v' * J .== v' * λ)
+        @test all(J * v  .== λ * v )
+        @test all(J * v' .== λ * v')
+        @test all(v  / J .== v  / λ)
+        @test all(v' / J .== v' / λ)
+        @test all(J \ v  .== λ \ v )
+        @test all(J \ v' .== λ \ v')
+    end
```

Comment:
What's the reason for not having simply `v * J == v * λ`? Equality of vectors is defined as equality of all elements, so I see no reason to write this all out.